### DISCUSSION TITLE
How to get the current time as an environment variable in actions?

### DISCUSSION THREAD

**Discussion by bmyjacks** [community/community#26342] [Actions]
<p>I hope to automatically create a new release through Actions and set the name to the current time, but I have failed many times. How can I do this?</p>

**Comment by bmyjacks** [comment]
<p>Just like this<br>
<img src="https://user-images.githubusercontent.com/38985658/181097203-e29bd015-c744-464e-a4a3-d1426ea976e6.png" alt="image" data-base62-sha1="pfFtBik9D85zJ03ugdilSvMqZKq" width="647" height="500"></p>

**Comment by airtower-luna** [comment]
<p>Assuming your build is running on Linux, you can just use the <code>date</code> command in a step, e.g:</p>
<pre><code class="lang-yaml">- name: set release date
  run: |
    echo "RELEASE_DATE=$(date --rfc-3339=date)" &gt;&gt; ${GITHUB_ENV}
</code></pre>
<p>That’ll give you the current date in YYYY-MM-DD format, adjust the <code>date</code> command line if you need another format.</p>

**Comment by bmyjacks** [comment]
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/3/8/38ad22d47e8fa27ac1da24eaecc06b5dbe22bcdf.png" data-download-href="/uploads/short-url/85nGclFnMAqH4VY42QiQkjTh0yr.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/38985658/181097206-ff74f0fb-fdca-44dd-aeca-b6d129ff45b6.png" alt="image" data-base62-sha1="85nGclFnMAqH4VY42QiQkjTh0yr" width="310" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/3/8/38ad22d47e8fa27ac1da24eaecc06b5dbe22bcdf_2_310x500.png, https://aws1.discourse-cdn.com/github/original/3X/3/8/38ad22d47e8fa27ac1da24eaecc06b5dbe22bcdf.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/3/8/38ad22d47e8fa27ac1da24eaecc06b5dbe22bcdf.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/3/8/38ad22d47e8fa27ac1da24eaecc06b5dbe22bcdf_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">406×653 30.9 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div><br>
It’s failed.<br>
Maybe my configuration is wrong?

**Comment by bmyjacks** [comment]
<p><img src="https://user-images.githubusercontent.com/38985658/181097207-d0a33898-9982-4c70-a92a-493170a8d44d.png" alt="image" data-base62-sha1="8YkGbIUzFmbX89J2mG9G1dXKqIn" width="668" height="151"></p>

**Comment by otter-computer** [comment, marked as answer]
<p>It looks like you’re sending the date into <code>${TIME}</code> instead of <code>$GITHUB_ENV</code>. To set an environment variable you need to send it to <code>$GITHUB_ENV</code> in the following format:</p>
<p><code>echo "{name}={value}" &gt;&gt; $GITHUB_ENV</code></p>
<aside class="onebox allowlistedgeneric">
  <header class="source">
      <img src="https://docs.github.com/assets/images/site/favicon.svg" class="site-icon" width="16" height="16">
      <a href="https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-commands-for-github-actions#setting-an-environment-variable" target="_blank" rel="noopener">docs.github.com</a>
  </header>
  <article class="onebox-body">
    <img src="" class="thumbnail" width="16" height="16">

<h3><a href="https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-commands-for-github-actions#setting-an-environment-variable" target="_blank" rel="noopener">Workflow commands for GitHub Actions - GitHub Docs</a></h3>



  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

**Comment by bmyjacks** [comment]
<p><img src="https://user-images.githubusercontent.com/38985658/181097208-6ce4059a-8ef5-4658-83a0-c4093dcd312c.png" alt="image" data-base62-sha1="dRJVtj6Jkugtp7WNI6z6zOMS5mF" width="672" height="369"><br>
It still have some problem.</p>

**Comment by bmyjacks** [comment]
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/2/b/2b99b489a87d8c913f42372cc823a793c58cbaee.png" data-download-href="/uploads/short-url/6dHQh5IOgJGMZOCCq4LLq1vHVvg.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/38985658/181097209-dd244380-21f1-437e-a7ee-d69a33dde46d.png" alt="image" data-base62-sha1="6dHQh5IOgJGMZOCCq4LLq1vHVvg" width="370" height="499" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/2/b/2b99b489a87d8c913f42372cc823a793c58cbaee_2_370x499.png, https://aws1.discourse-cdn.com/github/original/3X/2/b/2b99b489a87d8c913f42372cc823a793c58cbaee.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/2/b/2b99b489a87d8c913f42372cc823a793c58cbaee.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/2/b/2b99b489a87d8c913f42372cc823a793c58cbaee_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">451×608 30.5 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by otter-computer** [comment]
<p>You’ll likely need to wrap it in an expression syntax, otherwise it’ll just output as a string:</p>
<p><code>${{ env.TIME }}</code></p>

**Comment by bmyjacks** [comment]
<p>All right.It’s now working.<br>
Thank you so much! :grinning:<br>
<a class="mention" href="/airtower-luna">@airtower-luna</a> <a class="mention" href="/thomasshaped">@thomasshaped</a></p>

**Comment by orange723** [comment]
- run: |
    echo "time=$(date +%Y-%m-%d)" >> $GITHUB_ENV
        
echo ${{ env.time }}
